pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -377,12 +381,39 @@ func (p *Plugin) fetchAliasesPage(ctx context.Context, marker *string) (*string,
 			if err != nil {
 				return nil, err
 			}
-			p.log.Debug("Loaded key", keyIDTag, *alias.TargetKeyId, aliasTag, *alias.AliasName)
+			p.log.Debug("Key loaded", keyIDTag, *alias.TargetKeyId, aliasTag, *alias.AliasName)
 		}
 	}
 	return aliasesResp.NextMarker, nil
 }
 
+func (p *Plugin) scheduleDeleteTask(ctx context.Context) {
+	t := p.clock.Ticker(30 * time.Second)
+	defer t.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			p.log.Debug("Stopping schedule delete task", "reason", ctx.Err())
+			return
+		case <-t.C:
+			keyID, ok := <-p.scheduleDelete
+			if !ok {
+				continue
+			}
+			_, err := p.kmsClient.ScheduleKeyDeletion(ctx, &kms.ScheduleKeyDeletionInput{
+				KeyId:               aws.String(keyID),
+				PendingWindowInDays: aws.Int32(7),
```

Comment:
why 7 days?  may it be configurable?

Reply:
7 days is the least amount of days, while 30 is the max. I felt tempted to make it configurable but I'm also reluctant to add configurations to the plugins (unless they become desired or necessary). In this particular case the reasoning is that once the CA and keys rotates there is no reason (that I can think of) for keeping them longer than the minimal amount.

Reply:
> reluctant to add configurations to the plugins (unless they become desired or necessary)

💯